tests/support.cpp

```diff
@@ -82,4 +82,64 @@ TEST(DriveIndex, DriveZ)
 	EXPECT_EQ(25, drive_index('Z'));
 }
 
+TEST(Support_split, NoBoundingDelims)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split("a:/b:/c/d:/e/f/", ':'), expected);
+	EXPECT_EQ(split("a /b /c/d /e/f/", ' '), expected);
+}
+
+TEST(Support_split, DelimAtStartNotEnd)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split(":a:/b:/c/d:/e/f/", ':'), expected);
+	EXPECT_EQ(split(" a /b /c/d /e/f/", ' '), expected);
+}
+
+TEST(Support_split, DelimAtEndNotStart)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split("a:/b:/c/d:/e/f/:", ':'), expected);
+	EXPECT_EQ(split("a /b /c/d /e/f/ ", ' '), expected);
+}
+
+TEST(Support_split, DelimsAtBoth)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split(":a:/b:/c/d:/e/f/:", ':'), expected);
+	EXPECT_EQ(split(" a /b /c/d /e/f/ ", ' '), expected);
+}
+
+TEST(Support_split, MultiInternalDelims)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split("a:/b::/c/d:::/e/f/", ':'), expected);
+	EXPECT_EQ(split("a /b  /c/d   /e/f/", ' '), expected);
+}
+
+TEST(Support_split, MultiBoundingDelims)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split("::a:/b:/c/d:/e/f/:::", ':'), expected);
+	EXPECT_EQ(split("  a /b /c/d /e/f/   ", ' '), expected);
+}
+
+TEST(Support_split, MixedDelims)
+{
+	const std::vector<std::string> expected({"a", "/b", "/c/d", "/e/f/"});
+	EXPECT_EQ(split("::::a:/b::::/c/d:/e/f/", ':'), expected);
+	EXPECT_EQ(split("a /b  /c/d   /e/f/    ", ' '), expected);
+}
+
+TEST(Support_split, Empty)
+{
+	const std::vector<std::string> expected;
+	EXPECT_EQ(split("", ':'), expected);
+	EXPECT_EQ(split(":", ':'), expected);
+	EXPECT_EQ(split("::", ':'), expected);
+	EXPECT_EQ(split(" ", ' '), expected);
+	EXPECT_EQ(split("  ", ' '), expected);
+	EXPECT_EQ(split("   ", ' '), expected);
```

Comment:
I think this behaviour is incorrect, e.g. `split("::", ':')` should result in `{"", "", ""}`.

I believe you wanted to emulate Python's interface on how to handle string splits (good idea!), but python does it a little bit differently:
```python
>>> ':::'.split(':')
['', '', '', '']
>>> '   '.split(' ')
['', '', '', '']
>>> '   '.split()
[]

```

Reply:
If a string only contains delimiters, then there is no content when they're removed.

On the Python side, I suspect that returning a <None> would throw wrenches into nice Pythonic loops :-) Then again, they could return an empty vector and avoid the None. 

Also, pedantically, if a delimiter separates content on both its sides, then if:

"content_1:content_2" produces `{"content_1", "content_2"}`

And if the absence of delimited content still has value, then therefore:

 - ":content_2" should produce {"", "content_2"}
 - "content_1:" should produce {"content_1", ""}
 - ":" should produce {"", ""}
 - ":::" should produce 5 empties

The point being that if there's value in reporting non-content on the left-and-right sides of a delimiter, then that should be true even in a minimal case.

But more pragmatically -- the only value I see here is possibly counting the invalid or wasted delimeters that were used.  The split() implementation would get less elegant and the user-side would get uglier checking for empty() and discarding them too.

Maybe if/when we have a need to understand non-content we could add this?

Reply:
Actually, maybe we can implement it with a toggle that says "I care about non-content" versus "ignore non-content".

That said.. given the two scenarios - I honestly feel we only will ever care about the positive content and not the absence of content.

Reply:
Oh, no - the implementation will be *easier* than it is now. When looking for next string to extract start with next character instead of looking for next non-delimiter. One-line change.

I suggest adding another version of this function, one that does *not* accept delimiter as parameter (assuming delimiter is any whitespace), and acts as the current implementation (matching Python's `str.split()` - without parameters).

Reply:
Thanks @dreamer ; will do!